tests/test_integration.py

```diff
@@ -0,0 +1,520 @@
+        == """
+Failed to construct config file:
+Invalid file format for riotfile. Expected file with .py extension got 'rf'.
+""".lstrip()
+    )
+    assert result.returncode == 1
+
+    rf_path = tmp_path / "riotfile.py"
+    rf_path.write_text(
+        """
+from riot import Venv
+venv = Venv()typo1234
+""",
+    )
+    result = run("riot --file riotfile.py list", cwd=tmp_path)
+    assert (
+        """
+Failed to construct config file:
+Failed to parse riotfile 'riotfile.py'.
+""".lstrip()
+        in result.stderr
+    )
+    assert (
+        """
+SyntaxError: invalid syntax
+""".lstrip()
+        in result.stderr
+    )
+    assert result.returncode == 1
+
+
+def test_help(tmp_path: pathlib.Path) -> None:
+    result = run("riot --help", cwd=tmp_path)
+    assert (
+        result.stdout
+        == """
+Usage: riot [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -f, --file PATH  [default: riotfile.py]
+  -v, --verbose
+  -d, --debug
+  --version        Show the version and exit.
+  --help           Show this message and exit.
+
+Commands:
+  generate  Generate base virtual environments.
+  list      List all virtual env instances matching a pattern.
+  run       Run virtualenv instances with names matching a pattern.
+""".lstrip()
+    )
+    assert result.stderr == ""
+    assert result.returncode == 0
+
+
+def test_version(tmp_path: pathlib.Path) -> None:
+    result = run("riot --version", cwd=tmp_path)
+    assert result.stdout.startswith("riot, version ")
+    assert result.stderr == ""
+    assert result.returncode == 0
+
+
+def test_list_no_file_empty_file(tmp_path: pathlib.Path) -> None:
+    result = run("riot list", cwd=tmp_path)
+    assert (
+        result.stderr
+        == """
+Usage: riot [OPTIONS] COMMAND [ARGS]...
+Try 'riot --help' for help.
+
+Error: Invalid value for '-f' / '--file': Path 'riotfile.py' does not exist.
+""".lstrip()
+    )
+    assert result.returncode == 2
+
+    rf_path = tmp_path / "riotfile.py"
+    rf_path.write_text(
+        """
+from riot import Venv
+""",
+    )
+    result = run("riot list", cwd=tmp_path)
+    assert result.stderr == ""
+    assert result.stdout == ""
+    assert result.returncode == 0
+
+
+def test_list_configurations(tmp_path: pathlib.Path) -> None:
+    rf_path = tmp_path / "riotfile.py"
+    rf_path.write_text(
+        """
+from riot import Venv
+venv = Venv(
+    name="test",
+    pys=[3],
+    command="echo hi",
+)
+""",
+    )
+    result = run("riot list", cwd=tmp_path)
+    assert result.stderr == ""
+    assert result.stdout == "test  Python Interpreter(_hint='3')\n"
+    assert result.returncode == 0
+
+    rf_path.write_text(
+        """
+from riot import Venv
+venv = Venv(
+    name="test",
+    pys=[3],
+    command="echo hi",
+    pkgs={
+        "pkg1": ["==1.0", "==2.0"],
+    }
+)
+""",
+    )
+    result = run("riot list", cwd=tmp_path)
+    assert result.stderr == ""
+    assert re.search(
+        r"""
+test  .* 'pkg1==1.0'
+test  .* 'pkg1==2.0'
```

Comment:
We should know this exact string no? since this is the python version and we know the currently running version?

Reply:
The reasoning here is that the interpreter could be platform dependent. Right now it'd be fine because the output there would probably just be `Interpreter(_hint="3")` but in #137 we'd also have the path to the interpreter.

What we can do is at least match on `Interpreter(.*)` or similar. WDYT?

Reply:
Makes me wonder if we need better output here than what we have now `repr(Interpreter)`

could make this more of a consistent format

Reply:
not sure I am suggesting a specific change right now, just talking out loud.